### ISSUE TITLE
configuration doesn't detect missing TK* and DRAWEXE libraries for mechanics (OpenCascade)

### ISSUE THREAD

**Issue by radarsat1** [siconos/siconos#84]
Another error configuring my new 16.10 container, after cmake apparenty succeeds, I still get a build error:

```
[ 91%] Swig source
[ 91%] Linking CXX shared library libsiconos_mechanics.so
/usr/bin/ld: cannot find -lTKDraw
/usr/bin/ld: cannot find -lTKTopTest
/usr/bin/ld: cannot find -lTKViewerTest
/usr/bin/ld: cannot find -lTKXSDRAW
/usr/bin/ld: cannot find -lTKDCAF
/usr/bin/ld: cannot find -lTKXDEDRAW
/usr/bin/ld: cannot find -lTKTObjDRAW
/usr/bin/ld: cannot find -lTKQADraw
/usr/bin/ld: cannot find -lDRAWEXE
collect2: error: ld returned 1 exit status
mechanics/CMakeFiles/mechanics.dir/build.make:1459: recipe for target 'mechanics/libsiconos_mechanics.so.4.1.0' failed
make[2]: *** [mechanics/libsiconos_mechanics.so.4.1.0] Error 1
make[2]: Leaving directory '/home/siconos/build'
CMakeFiles/Makefile2:968: recipe for target 'mechanics/CMakeFiles/mechanics.dir/all' failed
make[1]: *** [mechanics/CMakeFiles/mechanics.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

I had installed liboce-_-dev packages, but evidently just missing some more OpenCascade packages.  CMake should have caught that.
Simple fix is just to `apt-get install oce-draw` and *re-run CMake_ (this last is necessary!)

**Comment by radarsat1** [comment]
I take it back, after re-running cmake and building again, I still get an error,

~~~
[ 91%] Building CXX object control/swig/CMakeFiles/_controller.dir/__/__/wrap/siconos/control/controllerPYTHON_wrap.cxx.o
[ 91%] Linking CXX shared library libsiconos_mechanics.so
/usr/bin/ld: cannot find -lDRAWEXE
collect2: error: ld returned 1 exit status
mechanics/CMakeFiles/mechanics.dir/build.make:1469: recipe for target 'mechanics/libsiconos_mechanics.so.4.1.0' failed
make[2]: *** [mechanics/libsiconos_mechanics.so.4.1.0] Error 1
make[2]: Leaving directory '/home/siconos/build'
CMakeFiles/Makefile2:968: recipe for target 'mechanics/CMakeFiles/mechanics.dir/all' failed
make[1]: *** [mechanics/CMakeFiles/mechanics.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
~~~

**Comment by radarsat1** [comment]
I should mention that I erased the build directory so the cmake cache is clear.  Still trying to figure out how to satisfy -lDRAWEXE.  As far as I can tell the `oce-draw` package only has an executable `/usr/bin/DRAWEXE`, but no library by that name..

**Comment by radarsat1** [comment]
I seems that I had WITH_MECHANISMS=ON and WITH_OCC=OFF.

I found that in `siconos_find_common_dep.cmake` there is the following:

~~~
# --- OCC ---
IF(WITH_OCC)
  if(NOT WITH_MECHANISMS)
    COMPILE_WITH(OCE 0.15 REQUIRED ONLY mechanics)
    SET(HAVE_OCC TRUE)
    LIST(REMOVE_ITEM SICONOS_LINK_LIBRARIES DRAWEXE)
  endif()
ENDIF()
~~~

So I thought by enabling WITH_OCC, it would compile successfully.  Unfortunately, same error with `-lDRAWEXE` as above.

**Comment by radarsat1** [comment]
Ok, sorry, I followed the above logic properly and changed to WITH_OCC=ON and WITH_MECHANISMS=OFF.

So I guess the question is, should we set WITH_MECHANISMS=OFF in siconos_default.cmake?  Because that's how I ended up here.  I copied `siconos_default.cmake` and only changed WITH_BULLET=ON.

Update: Nope, even with this config it's still trying to link DRAWEXE.

**Comment by radarsat1** [comment]
For now I set WITH_OCC=OFF and WITH_MECHANISMS=OFF, now it compiles but without OpenCascade.

**Comment by vacary** [comment]
For me it compiles when I set WITH_OCC=OFF and WITH_MECHANISMS=ON but I use my own compiled version of OCE.

Perhaps we have to ensure that the options are compatible. The fact that were search for DRAWEXE is not really suitable. Indeed, we copy and paste some cake configuration file bu we don't need DRAWEXE. 

I will try to remove it from the list of components.

**Comment by fperignon** [comment]
Hi,
This DRAWEXE thing comes from OCEConfig.cmake file, which is distributed with ubuntu/debian packages. 
In siconos_find_common_dep, drawexe is removed from the links list in both cases (WITH_OCC=ON or
WITH_MECHANISMS=ON but only if one version is less than 0.16). 
I think you should compile with WITH_MECHANISMS=ON. I tried it on ubuntu 16 
and it works. My oce packages are:
liboce-modeling-dev liboce-foundation-dev liboce-ocaf-dev liboce-visualization-dev oce-draw

Notice that I had to deactivate bullet since a file BulletBroadphase.hpp is missing. Is it suppose to come with bullet or with siconos/mechanics-proposed?

**Comment by radarsat1** [comment]
BulletBroadphase.hpp is an old filename that has been changed, I think it's still referenced in a test so I'll fix that.  I'll try again with WITH_MECHANISMS.  In any case maybe we could just remove DRAWEXE regardless of conditions?  I'll see how it goes.

**Comment by radarsat1** [comment]
For me what finally fixed this problem was removing DRAWEXE from mechanics_LINK_LIBRARIES, see d3a7864782d5874d7c78814837624244b53c2d1c

Not sure why, but it DRAWEXE simply was not in SICONOS_LINK_LIBRARIES.  It is added again if I have WITH_MECHANISMS turned on, and I have to add similar REMOVE_ITEM commands to the WITH_MECHANISMS section just below.

**Comment by vacary** [comment]
It seems the problem has been fixed in new release of oce:
https://github.com/tpaviot/oce/issues/660
For the moment, the package shipped in ubuntu 16 (0.17) add DRAWEXE has a library. I've tried to compile with 
- WITH_OCC=ON and WITH_MECHANISMS=ON, and
- WITH_OCC=ON and WITH_MECHANISMS=OFF
and I have the same problem.

**Comment by vacary** [comment]
Fixed with c62a01743220a1daca4ca742d066248ed40f1fcd
I think we can now implement a configuration  in CI with OCE as it is required in #30 under Ubuntu 16.04
with this list of packages
liboce-modeling-dev liboce-foundation-dev liboce-ocaf-dev liboce-visualization-dev oce-draw
